palette/src/okhsl.rs

```diff
@@ -0,0 +1,388 @@
+use crate::ok_utils::{toe, ChromaValues};
+use crate::white_point::D65;
+use crate::{
+    angle::{FromAngle, RealAngle},
+    convert::FromColorUnclamped,
+    num::{Arithmetics, Cbrt, IsValidDivisor, MinMax, One, Real, Trigonometry, Zero},
+    stimulus::{FromStimulus, Stimulus},
+    HasBoolMask, Oklab, OklabHue,
+};
+
+mod alpha;
+mod properties;
+#[cfg(feature = "random")]
+mod random;
+
+/// A Hue/Saturation/Lightness representation of [`Oklab`] in the `sRGB` color space.
+///
+/// Allows
+/// * changing hue/chroma/saturation, while keeping perceived lightness constant (like HSLuv)
+/// * changing lightness/chroma/saturation, while keeping perceived hue constant
+/// * changing the perceived saturation (more or less) proportionally with the numerical
+/// amount of change (unlike HSLuv)
+#[derive(Debug, Copy, Clone, ArrayCast, FromColorUnclamped, WithAlpha)]
+#[cfg_attr(feature = "serializing", derive(Serialize, Deserialize))]
+#[palette(
+    palette_internal,
+    white_point = "D65",
+    component = "T",
+    skip_derives(Oklab)
+)]
+#[repr(C)]
+pub struct Okhsl<T = f32> {
+    /// The hue of the color, in degrees of a circle, where for all `h`: `h+n*360 ==  h`.
+    ///
+    /// For fully saturated, bright colors
+    /// * 0° corresponds to a kind of magenta-pink (RBG #ff0188),
+    /// * 90° to a kind of yellow (RBG RGB #ffcb00)
+    /// * 180° to a kind of cyan (RBG #00ffe1) and
+    /// * 240° to a kind of blue (RBG #00aefe).
+    ///
+    /// For s == 0 or v == 0, the hue is irrelevant.
+    #[palette(unsafe_same_layout_as = "T")]
+    pub hue: OklabHue<T>,
+
+    /// The saturation (freedom of black or white) of the color.
+    ///
+    /// * `0.0` corresponds to pure mixture of black and white without any color.
+    /// The black to white relation depends on v.
+    /// * `1.0` to a fully saturated color without any white.
+    ///
+    /// For v == 0 the saturation is irrelevant.
+    pub saturation: T,
+
+    /// The relative luminance of the color, where  
+    /// * `0.0` corresponds to pure black
+    /// * `1.0` corresponds to white
+    ///
+    /// This luminance is visually similar to [Cielab](crate::Lab)'s luminance for a
+    /// `D65` reference white point.
+    ///
+    /// `Okhsv`'s `value` component goes from black to non-black
+    /// -- a maximally bright color in the `sRGB` gamut.
+    ///
+    /// `Okhsl`'s `lightness` component goes from black to white in the `sRGB` color space.
+    pub lightness: T,
+}
+
+impl<T> Okhsl<T> {
+    /// Create an Okhsl color.
+    pub fn new<H: Into<OklabHue<T>>>(hue: H, saturation: T, lightness: T) -> Self {
+        Self {
+            hue: hue.into(),
+            saturation,
+            lightness,
+        }
+    }
+
+    /// Create an `Okhsl` color. This is the same as `Okhsl::new` without the
+    /// generic hue type. It's temporary until `const fn` supports traits.
+    pub fn new_const(hue: OklabHue<T>, saturation: T, lightness: T) -> Self {
```

Comment:
Ah, you forgot to specify them as `const`. This seem to bee the case for all of them.
```suggestion
    pub const fn new_const(hue: OklabHue<T>, saturation: T, lightness: T) -> Self {
```